lib/puppet/provider/mount/parsed.rb

```diff
@@ -178,7 +178,21 @@ def to_line(result)
       end
     end
   else
-    record_line self.name, :fields => @fields, :separator => /\s+/, :joiner => "\t", :optional => optional_fields
+    record_line self.name, :fields => @fields, :separator => /\s+/, :joiner => "\t", :optional => optional_fields, :block_eval => :instance do
+      def pre_gen(record)
+        if !record[:options] || record[:options].empty?
+          if Facter.value(:kernel) == 'Linux'
+            record[:options] = 'defaults'
+          else
+            raise RuntimeError, "Mount[#{record[:name]}]: Field 'options' is required"
+          end
+        end
+        if !record[:fstype] || record[:fstype].empty?
+          raise RuntimeError, "Mount[#{record[:name]}]: Field 'fstype' is required"
```

Comment:
I am not sure about this one, but I guess `self.fail` is the best way to raise and error here because it will automatically add context like classname and failing resource. But haven't tried it